Orleans.Providers.MongoDB/StorageProviders/Serializers/JsonGrainStateSerializer.cs

```diff
@@ -10,34 +12,22 @@ public class JsonGrainStateSerializer : IGrainStateSerializer
     {
         private readonly JsonSerializer serializer;
 
-        public JsonGrainStateSerializer(IServiceProvider serviceProvider, MongoDBGrainStorageOptions options)
+        public JsonGrainStateSerializer(IOptions<JsonGrainStateSerializerOptions> options, IServiceProvider serviceProvider)
         {
             var jsonSettings = OrleansJsonSerializerSettings.GetDefaultSerializerSettings(serviceProvider);
```

Comment:
Can we not inject the settings?

Reply:
Can you elaborate on that please?

Reply:
If you're referring to `MongoDBGrainStorageOptions`, it is now agnostic to the kind of serializer used, which is why `ConfigureJsonSerializerSettings` property has been replaced by `GrainStateSerializer` and `JsonGrainStateSerializerOptions` is used as for configuring `JsonGrainStateSerializer`.

Reply:
I was referring to ServiceProvider. But it is fine for me